### ISSUE TITLE
Use other metrics to explain predictions in XGBoost

### ISSUE THREAD

**Issue by vlavorini** [TeamHG-Memex/eli5#254]
Hello,

it will be possible to include the other two metrics available in XGBoost, other than the gain?
I mean the weights and the coverage:

`https://xgboost.readthedocs.io/en/latest/python/python_api.html` , see section `get_score`

Thank you for the reply!

**Comment by lopuhin** [comment]
Hello @vlavorini,

If I understood you correctly, they are already supported, please see http://eli5.readthedocs.io/en/latest/libraries/xgboost.html - you can pass  optional argument `importance_type` to `explain_weights`.

**Comment by vlavorini** [comment]
So fast, just the time to go get a coffee, thank you!

Cool, I did not saw it because I was searching the availability of those metrics in `explain_prediction`, because that is the function that I need. 
And (if I'm seeing correct) there they are not implemented. Did you plan to implement them also in `explain_prediction`?

(this maybe deserve a separate post)
What is the reason to have chosen the gain instead of any other metric?
As opposite, the related default value in XGBoost's `plot_importance` method is the weight/frequency.

Thank you again!

**Comment by lopuhin** [comment]
Aha, I see - actually explain_prediction uses a special method of explanation which was not available in xgboost until very recently, it's not one of these three methods that you originally quoted. From ELI5 docs:

> Method for determining feature importances follows an idea from http://blog.datadive.net/interpreting-random-forests/. Feature weights are calculated by following decision paths in trees
of an ensemble. Each leaf has an output score, and expected scores can also be assigned
to parent nodes. Contribution of one feature on the decision path is how much expected score
changes from parent to child. Weights of all features sum to the output score of the estimator.

I think that the most recent XGBoost release also has SHAP support: https://github.com/slundberg/shap#xgboost-lightgbm-example but we didn't yet have time to evaluate it and compare to what ELI5 implements.

**Comment by vlavorini** [comment]
Thank you, your information has been very useful!

**Comment by lopuhin** [comment]
Glad that it helped!

**Comment by lopuhin** [comment]
Although SHAP support in eli5 probably makes sense to at least investigate.

**Comment by lopuhin** [comment]
Duplicate of #240
